x-pack/plugins/fleet/server/services/epm/packages/get_template_inputs.ts

```diff
@@ -128,7 +181,146 @@ export async function getTemplateInputs(
         sortKeys: _sortYamlKeys,
       }
     );
-    return yaml;
+    return addCommentsToYaml(yaml, buildIndexedPackage(packageInfo));
   }
+
   return { inputs: [] };
 }
+
+function getPlaceholder(varDef: RegistryVarsEntry) {
+  return `<${varDef.name.toUpperCase()}>`;
+}
+
+function addPlaceholderIfNeeded(
+  varDef: RegistryVarsEntry,
+  varValue: PackagePolicyConfigRecordEntry
+) {
+  const placeHolder = `<${varDef.name.toUpperCase()}>`;
+  if (varDef && !varValue.value && varDef.type !== 'yaml') {
+    varValue.value = placeHolder;
+  } else if (varDef && varValue.value && varValue.value.length === 0 && varDef.type === 'text') {
+    varValue.value = [placeHolder];
+  }
+}
+
+function buildIndexedPackage(packageInfo: PackageInfo): PackageWithInputAndStreamIndexed {
+  return (
+    packageInfo.policy_templates?.reduce<PackageWithInputAndStreamIndexed>(
+      (inputsAcc, policyTemplate) => {
+        const inputs = getNormalizedInputs(policyTemplate);
+
+        inputs.forEach((packageInput) => {
+          const inputId = `${policyTemplate.name}-${packageInput.type}`;
+
+          const streams = getStreamsForInputType(
+            packageInput.type,
+            packageInfo,
+            isIntegrationPolicyTemplate(policyTemplate) && policyTemplate.data_streams
+              ? policyTemplate.data_streams
+              : []
+          ).reduce<
+            Record<
+              string,
+              RegistryStream & {
+                data_stream: { type: string; dataset: string };
+              }
+            >
+          >((acc, stream) => {
+            const streamId = `${packageInput.type}-${stream.data_stream.dataset}`;
+            acc[streamId] = {
+              ...stream,
+            };
+            return acc;
+          }, {});
+
+          inputsAcc[inputId] = {
+            ...packageInput,
+            streams,
+          };
+        });
+        return inputsAcc;
+      },
+      {}
+    ) ?? {}
+  );
+}
+
+function addCommentsToYaml(
+  yaml: string,
+  packageIndexInputAndStreams: PackageWithInputAndStreamIndexed
+) {
+  const doc = yamlDoc.parseDocument(yaml);
+  // Add input and streams comments
+  const yamlInputs = doc.get('inputs');
+  if (yamlDoc.isCollection(yamlInputs)) {
+    yamlInputs.items.forEach((inputItem) => {
+      if (!yamlDoc.isMap(inputItem)) {
+        return;
+      }
+      const inputIdNode = inputItem.get('id', true);
+      if (!yamlDoc.isScalar(inputIdNode)) {
+        return;
+      }
+      const inputId = inputIdNode.value as string;
+      const pkgInput = packageIndexInputAndStreams[inputId];
+      if (pkgInput) {
+        inputItem.commentBefore = ` ${pkgInput.title} ${
+          pkgInput.description ? ` ${pkgInput.description}` : ''
```

Comment:
```suggestion
        inputItem.commentBefore = ` ${pkgInput.title}${
          pkgInput.description ? `: ${pkgInput.description}` : ''
```

would be good to have a separator here, either like this or a new line break